modules/data.atmosphere/R/tdm_generate_subdaily_models.R

```diff
@@ -0,0 +1,226 @@
+    for (j in seq_along(vars.info$CF.name)) {
+        if (exists(as.character(vars.info$CF.name[j]), tem$var) == FALSE) {
+            dat.train[[j]] <- NA
+        } else {
+            dat.train[[j]] <- ncdf4::ncvar_get(tem, as.character(vars.info$CF.name[j]))
+        }
+    }
+    names(dat.train) <- vars.info$CF.name
+    dat.train <- data.frame(dat.train)
+    
+    dat.train$date <- strptime(paste(dat.train$year, dat.train$doy + 1, 
+        dat.train$hour, sep = "-"), "%Y-%j-%H", tz = "GMT")
+    dat.train$time.hr <- as.numeric(difftime(dat.train$date, paste0((min(dat.train$year) - 
+        1), "-12-31 23:00:00"), tz = "GMT", units = "hour"))
+    dat.train$time.day <- as.numeric(difftime(dat.train$date, paste0((min(dat.train$year) - 
+        1), "-12-31 23:00:00"), tz = "GMT", units = "day")) - 1/24
+    dat.train$time.day2 <- as.integer(dat.train$time.day + 1/(48 * 2)) + 
+        1  # Offset by half a time step to get time stamps to line up
+    dat.train <- dat.train[order(dat.train$time.hr), ]
+    
+    
+    # ----- 1.1 Coming up with the daily means that are what we can
+    # use as predictors ----------
+    
+    train.day <- aggregate(dat.train[, c("air_temperature", "precipitation_flux", 
+        "surface_downwelling_shortwave_flux_in_air", "surface_downwelling_longwave_flux_in_air", 
+        "air_pressure", "specific_humidity", "wind_speed")], by = dat.train[, 
+        c("year", "doy")], FUN = mean)
+    names(train.day)[3:9] <- c("air_temperature_mean.day", "precipitation_flux.day", 
+        "surface_downwelling_shortwave_flux_in_air.day", "surface_downwelling_longwave_flux_in_air.day", 
+        "air_pressure.day", "specific_humidity.day", "wind_speed.day")
+    train.day$air_temperature_max.day <- aggregate(dat.train[, c("air_temperature")], 
+        by = dat.train[, c("year", "doy")], FUN = max)$x
+    train.day$air_temperature_min.day <- aggregate(dat.train[, c("air_temperature")], 
+        by = dat.train[, c("year", "doy")], FUN = min)$x
+    summary(train.day)
+    
+    
+    dat.train <- merge(dat.train[, ], train.day, all.x = T, all.y = T)
+    summary(dat.train)
+    
+    # ----- 1.2 Setting up a 1-hour lag -- smooth transitions at
+    # midnight NOTE: because we're filtering from the present back through
+    # the past, -1 will associate the closest hour that we've already done
+    # (midnight) with the day we're currently working on ----------
+    vars.hour <- c("air_temperature", "precipitation_flux", "surface_downwelling_shortwave_flux_in_air", 
+        "surface_downwelling_longwave_flux_in_air", "air_pressure", "specific_humidity", 
+        "wind_speed")
+    vars.lag <- c("lag.air_temperature", "lag.precipitation_flux", "lag.surface_downwelling_shortwave_flux_in_air", 
+        "lag.surface_downwelling_longwave_flux_in_air", "lag.air_pressure", 
+        "lag.specific_humidity", "lag.wind_speed")
+    lag.day <- dat.train[dat.train$hour == 23, c("year", "doy", "time.day2", 
```

Comment:
This just allows us to select the previous value and begin the data.frame. Even if the training data is 6hr reanalysis, this should be fine since it will be grabbing the value from the previous row, and the times will match up in the date and hr steps